### PR TITLE
Add a light version of CI

### PR DESCRIPTION
**Pull request by szepeviktor**
The general stuff.
- quote all YAML string
- use indentation from YAML spec (empty line with dash separates steps - as braces in PSR-12)
- 0️⃣ nice keycap emojis

This is the _light_ version. [Many things excluded.
](https://github.com/szepeviktor/byte-level-care/blob/master/.github/workflows/back-end.yml)

### CODE REVIEW

File: .github/workflows/back-end.yml [hunk 1]
```diff
@@ -0,0 +1,228 @@
+# yaml-language-server: $schema=https://json.schemastore.org/github-workflow
```
Comment (szepeviktor) [inline .github/workflows/back-end.yml:1]: This is IDE-s. The schema makes autocomplete/validation possible.

File: .github/workflows/back-end.yml [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        runs-on: "ubuntu-22.04"
+        timeout-minutes: 1
+        steps:
+        -
+            name: "Checkout repository"
+            uses: "actions/checkout@v3"
+        -
+            name: "Check file permissions"
+            run: |
+                test "$(find ./ -type f -not -path './.git/*' -executable)" = ""
```
Comment (szepeviktor) [inline .github/workflows/back-end.yml:30]: When badly configured git clients/Windows user turn on the execute permission on files.
`chmod +x`

File: .github/workflows/back-end.yml [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+        -
+            name: "Find non-printable ASCII characters"
+            run: |
+                ! LC_ALL=C.UTF-8 git grep --perl-regexp --line-number -e '[^ -~]' -- '*\.php'
```
Comment (szepeviktor) [inline .github/workflows/back-end.yml:34]: Printable characters are enough to make the computer tick.

File: .github/workflows/back-end.yml [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+                php-version: "8.1"
+                coverage: "none"
+                tools: "parallel-lint"
+        -
+            name: "Checkout repository"
+            uses: "actions/checkout@v3"
+        -
+            name: "Search for conflict markers 🐌"
+            run: |
+                ! git grep --line-number -e '^\(<<<<<<<\s\|=======\s\|=======$\|>>>>>>>\s\||||||||\s\)'
```
Comment (szepeviktor) [inline .github/workflows/back-end.yml:54]: When you commit conflict markers :)

File: .github/workflows/back-end.yml [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+        -
+            name: "Search for invalid complex curly syntax 🐌"
+            run: |
+                ! git grep -e '\${[A-Z_a-z]' -- '*\.php'
```
Comment (szepeviktor) [inline .github/workflows/back-end.yml:58]: ❌ `${variable}`

File: .github/workflows/back-end.yml [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+        -
+            name: "Check source code for syntax errors"
+            run: "composer exec --no-interaction -- parallel-lint src/ tests/"
```
Comment (szepeviktor) [inline .github/workflows/back-end.yml:61]: Essential to be syntax error free.

File: .github/workflows/back-end.yml [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+        - "byte_level"
+        - "syntax_errors"
+        strategy:
+            #fail-fast: false
+            matrix:
+                php-version:
+                - "8.1"
+                - "8.2"
+                dependencies:
+                - "lowest"
```
Comment (szepeviktor) [inline .github/workflows/back-end.yml:75]: Run with lowest deps too.

File: .github/workflows/back-end.yml [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+            run: "composer run feature"
+        -
+            name: "Send coverage to Coveralls"
+            env:
+                COVERALLS_REPO_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+            if: "${{ env.COVERALLS_REPO_TOKEN && matrix.php-version == '8.2' && matrix.dependencies == 'highest' }}"
+            run: |
+                wget --no-verbose \
+                    "https://github.com/php-coveralls/php-coveralls/releases/download/v2.5.3/php-coveralls.phar"
+                #php ./php-coveralls.phar --verbose
```
Comment (szepeviktor) [inline .github/workflows/back-end.yml:111]: Add coverage in `phpunit.xml` and remove comment from here.
https://coveralls.io/github/PrinsFrank/MRTD

File: .github/workflows/back-end.yml [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+                coverage: "none"
+        -
+            name: "Checkout repository"
+            uses: "actions/checkout@v3"
+        -
+            name: "Check JSON files"
+            run: |
+                git ls-files --cached -z -- '*\.json' \
+                    | xargs -t --null -L 1 -- \
+                        php -r 'json_decode(file_get_contents($argv[1]), null, 512, JSON_THROW_ON_ERROR);'
```
Comment (szepeviktor) [inline .github/workflows/back-end.yml:135]: Need valid JSON files!

File: .github/workflows/back-end.yml [hunk 10]
```diff
@@ -0,0 +1,228 @@
...
+            name: "Validate Composer configuration"
+            run: "composer validate --no-interaction --strict"
+        -
+            name: "Install dependencies"
+            uses: "ramsey/composer-install@v2"
+            with:
+                dependency-versions: "highest"
+        -
+            name: "Check PSR-4 mapping 🐌"
+            run: "composer dump-autoload --optimize --strict-psr"
```
Comment (szepeviktor) [inline .github/workflows/back-end.yml:146]: When FQCN does not match the filesystem path 💣

File: .github/workflows/back-end.yml [hunk 11]
```diff
@@ -0,0 +1,228 @@
...
+        -
+            name: "Check EditorConfig configuration"
+            run: "test -f .editorconfig"
+        -
+            name: "Check adherence to EditorConfig"
+            uses: "greut/eclint-action@v0"
+        -
+            name: "Look for TAB characters in the middle of the line 🐌"
+            run: |
+                ! git grep --perl-regexp --line-number -I '^(?!//)[^\t]+\t'
```
Comment (szepeviktor) [inline .github/workflows/back-end.yml:181]: Weird things do happen: TAB characters in the middle of the line.

File: .github/workflows/back-end.yml [hunk 12]
```diff
@@ -0,0 +1,228 @@
...
+            with:
+                dependency-versions: "highest"
+        -
+            name: "Detect coding standards violations"
+            id: "phpcsfixer"
+            run: "composer run cs"
+        -
+            name: "Annotate PHP-CS-Fixer results"
+            if: "${{ failure() && steps.phpcsfixer.outcome == 'failure' }}"
+            run: "composer run cs -- --format=checkstyle | cs2pr"
```
Comment (szepeviktor) [inline .github/workflows/back-end.yml:194]: Add nice GitHub Annotations.

File: .github/workflows/back-end.yml [hunk 13]
```diff
@@ -0,0 +1,228 @@
...
+        -
+            # Move TODO-s into GitHub issues!
```
Comment (szepeviktor) [inline .github/workflows/back-end.yml:196]: No TODO-s!! 👉🏻 GitHub issues 😍

File: .github/workflows/back-end.yml [hunk 14]
```diff
@@ -0,0 +1,228 @@
...
+        runs-on: "ubuntu-22.04"
+        timeout-minutes: 1
+        steps:
+        -
+            name: "Checkout repository"
+            uses: "actions/checkout@v3"
+        -
+            name: "Check for ignored files in the index 🐌"
+            run: |
+                IGNORED_FILES="$(git ls-files --cached --ignored --exclude-standard)"
```
Comment (szepeviktor) [inline .github/workflows/back-end.yml:215]: When I commit ignored files ⚡

File: .github/workflows/back-end.yml [hunk 15]
```diff
@@ -0,0 +1,228 @@
...
+                test -z "${IGNORED_FILES}"
+        -
+            name: "Check exported files"
+            run: |
+                EXPECTED="LICENSE,README.md,composer.json"
+                CURRENT="$(
+                    git archive HEAD \
```
Comment (szepeviktor) [inline .github/workflows/back-end.yml:222]: Check contents of GitHub ZIP-s (Packagist ...)

File: .editorconfig [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+
+[*.md]
+max_line_length = 300
+trim_trailing_whitespace = false
+
+[.github/workflows/back-end.yml]
+max_line_length = 165
+
+[*.php]
+max_line_length = 300
```
Comment (szepeviktor) [inline .editorconfig:25]: Please consider writing shorter lines ~120

**Comment by szepeviktor** [comment]
All right! 🐌 Snaily show very interesting stuff.

File: .github/workflows/back-end.yml (hunk 4 above)
Comment (PrinsFrank) [reply .github/workflows/back-end.yml:54]: Nice!

File: .github/workflows/back-end.yml (hunk 10 above)
Comment (PrinsFrank) [reply .github/workflows/back-end.yml:146]: didn't know this option existed, great!

File: .editorconfig (hunk 16 above)
Comment (PrinsFrank) [reply .editorconfig:22]: Haha, will do! It's been a while since I've had that as a standard as I'd like to read as much as possible in my viewport. ;)

**Review by PrinsFrank — APPROVED** [review]

**Comment by PrinsFrank** [comment]
Thank you so much! Learned about a lot of new shiny toys here!

**Comment by szepeviktor** [comment]
Glad to contribute.

**Pull request merged**